### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by skriss**
- updates github.com/vmware-tanzu/velero
- updates k8s deps to match velero
- updates Azure SDK to v42.0.0 (latest)

The primary purpose of the updates is to get the new Velero plugin server code that (a) binds the `--features` flag; and (b) ignores unknown flags during flag parsing.

I did a quick test of the ObjectStore and VolumeSnapshotter and it looked fine.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -55,11 +55,11 @@
```
Comment (ashish-amarnath) [inline Gopkg.toml:55]: Should we pin the velero to a specific version?

Comment (skriss) [reply Gopkg.toml:55]: we can definitely do that just after velero 1.4 is out, before releasing a new version of this plugin. Until then, there is no version of velero that has the changes that we want to integrate/test with.

Comment (ashish-amarnath) [reply Gopkg.toml:55]: Sounds good!
I will also convert this to switch to go modules for all the plugins.

Comment (skriss) [reply Gopkg.toml:55]: cool, yeah also wanted to do that but didn't want to take the time right now. I'm planning to create issues in all the plugin repos for (a) go module migration, and (b) refactoring the repo org/makefile/dockerfile to be closer to what's in the example repo

**Pull request merged**